gustaf/helpers/notebook.py

```diff
@@ -0,0 +1,123 @@
+import importlib
+
+import numpy as np
+import vedo
+from IPython.display import display
+from ipywidgets import GridspecLayout
+
+
+def get_shape(N, x, y):
+    """Taken verbatim from vedo plotter:show function.
+
+    Args:
+        N (_type_): _description_
+        x (_type_): _description_
+        y (_type_): _description_
+
+    Returns:
+        _type_: _description_
+    """
+    nx = int(np.sqrt(int(N * y / x) + 1))
+    ny = int(np.sqrt(int(N * x / y) + 1))
+    lm = [
+        (nx, ny),
+        (nx, ny + 1),
+        (nx - 1, ny),
+        (nx + 1, ny),
+        (nx, ny - 1),
+        (nx - 1, ny + 1),
+        (nx + 1, ny - 1),
+        (nx + 1, ny + 1),
+        (nx - 1, ny - 1),
+    ]
+    ind, minl = 0, 1000
+    for i, m in enumerate(lm):
+        l_something = m[0] * m[1]
+        if N <= l_something < minl:
+            ind = i
+            minl = l_something
+    return lm[ind]
+
+
+class K3DPlotterN(GridspecLayout):
+    """Helper to plot in notebooks with k3d.
+
+    Sets up K3D plotter especially if multiple plots are to be shown in the
+    notebook.
+    """
+
+    def __init__(self, N, size, background=0xFFFFFF):
+        """Setups the plotter with potentially multiple plots.
+
+        Parameters
+        ----------
+        N: int
+            Number of plots to be shown.
+        size: (str, list)
+            See vedo.Plotter for details.
+        background: ((color, str), optional)
+          See vedo.Plotter for details. Defaults to 0xFFFFFF.
+        """
+        if importlib.util.find_spec("k3d") is None:
+            raise ImportError(
+                "k3d is not installed. Please install it with `pip "
+                "install k3d`."
+            )
+        self.N = N
+        self.x, self.y = get_shape(N, *(2160, 1440))
```

Comment:
```suggestion
        self.x, self.y = get_shape(N, *(2160, 1440))
```
If I understood this right, I think this should take `*size` argument and at l.74, you'd give `size=[size[0] / self.x, size[1] / self.y]`

Reply:
As I said in the original function this is taken from the vedo implementation for multiple plotters in a single window. They used this as far as I know. I am not sure what the size variable is and if this can be used for this.